lib/_http_client.js

```diff
@@ -197,7 +197,9 @@ function ClientRequest(options, cb) {
     self = null;
   });
 
-  this._ended = false;
+  self._ended = false;
+  self.res = null;
+  self.aborted = 0;
```

Comment:
If we're setting this to zero now, we should both remove `ClientRequest.prototype.aborted = undefined;` and change the `if (this.aborted === undefined)` check to `if (!this.aborted)` below.